src/actions/Fetch.js

```diff
@@ -33,6 +34,9 @@ export class Fetch extends BaseAction {
     // Extend fetch options
     options = Object.assign({}, this.defaultOptions, options)
 
+    // Make sure params is an immutable map
+    params = !params || params && Immutable.Iterable.isIterable(params) ? params : Immutable.fromJS(params)
```

Comment:
For the params we want to convert back to raw js since these will be used for the fetch api request. 

I think we want something like:
```
params = Immutable.Iterable.isIterable(params) ? params.toJS() : params
```